exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -106,16 +128,18 @@ func newPRWExporter(cfg *Config, set exporter.Settings) (*prwExporter, error) {
 	userAgentHeader := fmt.Sprintf("%s/%s", strings.ReplaceAll(strings.ToLower(set.BuildInfo.Description), " ", "-"), set.BuildInfo.Version)
 
 	prwe := &prwExporter{
-		endpointURL:       endpointURL,
-		wg:                new(sync.WaitGroup),
-		closeChan:         make(chan struct{}),
-		userAgentHeader:   userAgentHeader,
-		maxBatchSizeBytes: cfg.MaxBatchSizeBytes,
-		concurrency:       cfg.RemoteWriteQueue.NumConsumers,
-		clientSettings:    &cfg.ClientConfig,
-		settings:          set.TelemetrySettings,
-		retrySettings:     cfg.BackOffConfig,
-		retryOnHTTP429:    retryOn429FeatureGate.IsEnabled(),
+		endpointURL:         endpointURL,
+		wg:                  new(sync.WaitGroup),
+		closeChan:           make(chan struct{}),
+		userAgentHeader:     userAgentHeader,
+		maxBatchSizeBytes:   cfg.MaxBatchSizeBytes,
+		concurrency:         cfg.RemoteWriteQueue.NumConsumers,
+		clientSettings:      &cfg.ClientConfig,
+		settings:            set.TelemetrySettings,
+		retrySettings:       cfg.BackOffConfig,
+		retryOnHTTP429:      retryOn429FeatureGate.IsEnabled(),
+		enableSendingRW2:    enableSendingRW2FeatureGate.IsEnabled(),
+		RemoteWriteProtoMsg: cfg.RemoteWriteProtoMsg,
```

Comment:
Instead of setting `cfg.RemoteWriteProtoMsg` directly, you should set `RemoteWriteProtoMsg` if `!enableSendingRW2`. That will save you from having to check the same elsewhere in the code, and things will be a lot simpler.

Reply:
Do we want to go with this approach or this one David suggested in another comment?
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35888#discussion_r1820700091

Reply:
Do we think we will ever support more than two protocols at once?  I don't think so... If that is the case, a single bool is OK here, which encompasses the user config and the feature gate.

Reply:
I think we'll support more than 2 protocols, yes. Mostly because PRW will continue to evolve and not all receivers will be able to implement new versions as they come up.